### ISSUE TITLE
[Question]: 

### ISSUE THREAD

**Issue by ankitkumar1302** [danielmiessler/fabric#560]
### What is your question?

I am facing this error how to solve it ??

Error: Error code: 404 - {'error': {'message': 'The model `gpt-4-turbo-preview` does not exist or you do not have access to it.', 'type': 'invalid_request_error', 'param': None, 'code': 'model_not_found'}}
Error code: 404 - {'error': {'message': 'The model `gpt-4-turbo-preview` does not exist or you do not have access to it.', 'type': 'invalid_request_error', 'param': None, 'code': 'model_not_found'}}

**Comment by WillJesperson** [comment]
did you input your api key? If not you can access it with this command
`nano ~/.config/fabric/.env`
and then you can copy and paste it in where indicated

**Comment by ankitkumar1302** [comment]
Yes I will input the api key but i am getting this error I will try this cmd to solve this error.

**Comment by kript** [comment]
TL;DR `DEFAULT_MODEL` is not set by default and the default in the code appears to not match what openai are offering.

---


I have the same issue - I only have OpenAI key configured with `--setup`.

```
$ fabric --setup
Welcome to Fabric. Let's get started.
Please enter your OpenAI API key. If you do not have one or if you have already entered it, press enter.
sk-proj-RESTOFKEY
Please enter your claude API key. If you do not have one, or if you have already entered it, press enter.


Please enter your Google API key. If you do not have one, or if you have already entered it, press enter.


Please enter your YouTube API key. If you do not have one, or if you have already entered it, press enter.


Updating patterns...
Downloaded zip file successfully.
Extracted zip file successfully.
Patterns updated successfully.
Creating empty environment file...
Environment file created.

$ fabric --listmodels
GPT Models:
gpt-3.5-turbo
gpt-3.5-turbo-0125
gpt-3.5-turbo-0301
gpt-3.5-turbo-0613
gpt-3.5-turbo-1106
gpt-3.5-turbo-16k
gpt-3.5-turbo-16k-0613
gpt-3.5-turbo-instruct
gpt-3.5-turbo-instruct-0914

Local Models:

Claude Models:

Google Models:

$ cat Notes.md | fabric --pattern summarize
Error: Error code: 404 - {'error': {'message': 'The model `gpt-4-turbo-preview` does not exist or you do not have access to it.', 'type': 'invalid_request_error', 'param': None, 'code': 'model_not_found'}}
Error code: 404 - {'error': {'message': 'The model `gpt-4-turbo-preview` does not exist or you do not have access to it.', 'type': 'invalid_request_error', 'param': None, 'code': 'model_not_found'}}

$ cat ~/.config/fabric/.env
#No API key set
OPENAI_API_KEY=sk-proj-RESTOFKEY
```

further checks reveals that this model is hardcoded;

```
git grep gpt-4-turbo-preview
installer/client/cli/utils.py:                self.model = 'gpt-4-turbo-preview'
```
This is set [here](https://github.com/danielmiessler/fabric/blob/6c0e8a9b3af76843307217bce2f63a139de2b0f3/installer/client/cli/utils.py#L54) 

The default model is indeed not set, however, the ReadME.md does not mention needing to set it _unless_ you are using a local model or similar.

```
set | grep DEFAULT_MODEL
```
when I set this, I get a different error;

```bash
export DEFAULT_MODEL=gpt-3.5-turbo
Error: Error code: 429 - {'error': {'message': 'You exceeded your current quota, please check your plan and billing details. For more information on this error, read the docs: https://platform.openai.com/docs/guides/error-codes/api-errors.', 'type': 'insufficient_quota', 'param': None, 'code': 'insufficient_quota'}}
Error code: 429 - {'error': {'message': 'You exceeded your current quota, please check your plan and billing details. For more information on this error, read the docs: https://platform.openai.com/docs/guides/error-codes/api-errors.', 'type': 'insufficient_quota', 'param': None, 'code': 'insufficient_quota'}}
```
which is odd, since I have a paid account on the Plus tier, but gives me something else to investigate.
